### PR TITLE
Add sol time

### PR DESCRIPTION
**Pull request by joanibal**
## Purpose
Adds solution time as an optional parameter when writing out files to tecplot.

Adds an option to recompute the data of a volume. This is useful when looking at how the embedding volume of an FFD has changed as the points move. 

It is related to this [pyGeo PR](https://github.com/mdolab/pygeo/pull/133)

## Expected time until merged
1 week
## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by nwu63 — COMMENTED** [review]
Looks good. A few questions:
- when would you want to recompute stuff? Why would it become inconsistent?
- do we want to make it consistent with pySurface and pyCurve, e.g. add the same `recompute` argument to all the `computeData` functions?

**Comment by joanibal** [comment]
- The data does not get recomputed after embedding since it is only used during the embedding. 

- Yeah, we probably should add the `recompute` arg to the the other classes too. I wish I didn't have to duplicate this code across each class, but I'm not about to restructure pyspyline.

**Comment by joanibal** [comment]
@nwu63 
I have added the recompute arg to the other data types now

**Review by nwu63 — DISMISSED** [review]

File: pyspline/pyCurve.py [hunk 1]
```diff
@@ -802,15 +802,15 @@ def projectCurveMultiSol(self, inCurve, nIter=25, eps=1e-10, **kwargs):
 
         return np.array(uSol), np.array(tSol), np.array(diff)
 
-    def computeData(self):
+    def computeData(self, recompute=False):
```
Comment (akleb) [inline pyspline/pyCurve.py:805]: Can you add a doc-string for the recompute flag? (this applies to the other `computeData` functions

**Review by akleb — COMMENTED** [review]
I am not entirely sure why the `recompute` flag is needed, presumably something outside the scope of this repo? I didn't see anything obvious in the matching pygeo PR either. It seems to me like whenever the function is called we would just expect it to recompute anyway, why was this not the case?

File: pyspline/pyCurve.py (hunk 1 above)
Comment (joanibal) [reply pyspline/pyCurve.py:805]: yep

**Comment by joanibal** [comment]
`computeData()` gets the data for the Greville points on the splined object (curve, surface, or volume). These points are used to find a good initial guess for the point projection routines. 
When we use the splined volume for an FFD we embed many points without changing the volume. So you don't need to recompute the Greville points for each point, so pySpline saves the data and doesn't recompute it after the first call. 
I can't say I totally agree with the naming and how the program was implemented, but I believe that was the original thinking. 

I use computeData(recompute=True) [here](https://github.com/joanibal/pygeo/blob/3ab177c6d75d14899034ab46f7578faba2233e42/pygeo/parameterization/DVGeo.py#L3947) in the related PR for pyGeo.  
If you are writing the embedding volume out at each iteration, then you need to recompute the Greville points since the volume has been deformed. This way you are writing out the updated embedding volume.

**Review by nwu63 — APPROVED** [review]

**Review by akleb — APPROVED** [review]

**Pull request merged**